### ISSUE TITLE
PyTorch freezes on second call to scripted densenet model from torchvision

### ISSUE THREAD

**Issue by BowenBao** [pytorch/pytorch#32751]
## 🐛 Bug

PyTorch freezes on second call to scripted densenet model from torchvision.

## To Reproduce

The script freezes after printing the first result.

```python
import torch
import torchvision

"""
Freezes on the second call to run the script model.
"""
model = torchvision.models.densenet121(pretrained=True)
x = torch.randn(2, 3, 224, 224, requires_grad=True)

script_model = torch.jit.script(model)
print('resuit:', script_model(x))

print('second result:', script_model(x))
```

## Environment

Both PyTorch and torchvision are nightly, installed by `conda install pytorch torchvision cpuonly -c pytorch-nightly`

PyTorch version: 1.5.0.dev20200128
Is debug build: No
CUDA used to build PyTorch: Could not collect

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.7
Is CUDA available: No
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 410.78
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.1
/usr/local/cuda-10.0/lib64/libcudnn.so.7

**Comment by suo** [comment]
@ZolotukhinM possibly the same subgraph slicing issue we are seeing?

**Comment by ZolotukhinM** [comment]
Quite likely

**Comment by ZolotukhinM** [comment]
Let me share some results of our investigation of a similar (or probably the same) issue.

Recently we started to see long compile-times on some models (I noticed that on resnet50), and after looking at time profiles I found that more than 90% of the time is spent in `createAutodiffSubgraphs` pass. One of the proposed mitigations was to make the subgraph slicer run only when the ops involved *must* be recording a gradient, while currently it runs when they *may* be recording a gradient.

It would be nice if we can verify whether this is the same issue or not.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
